java/memory/src/main/java/io/netty/buffer/ArrowBuf.java

```diff
@@ -403,89 +190,28 @@ public ArrowBuf slice(int index, int length) {
      * .html#wiki-h3-5, which
      * explains that derived buffers share their reference count with their parent
      */
-    final ArrowBuf newBuf = ledger.newArrowBuf(offset + index, length);
+    final ArrowBuf newBuf = referenceManager.deriveBuffer(this, index, length);
     newBuf.writerIndex(length);
     return newBuf;
   }
 
-  @Override
-  public ArrowBuf duplicate() {
-    return slice(0, length);
-  }
-
-  @Override
-  public int nioBufferCount() {
-    return 1;
-  }
-
-  @Override
   public ByteBuffer nioBuffer() {
```

Comment:
consider returning a collection here (if it isn't too big a change, even even we want to change to support 64 bit size memory in a separate PR).